### PR TITLE
Split Layouts

### PR DESCRIPTION
**Pull request by atomworks**
### Description

This PR is focusd around the addition of the `SplitLayout` component, a layout component that allows for a resizable split layout that can be horizontal or vertical, and also lets you nest the split layout for more advanced options.

The drag handles allow the second area of the layout to be optionally hideable. The order can also be reversed to place the hideable content pane on the left instead. The minimum width setting will enforce a minimum container size, after which there is a point where continuing to try and shrink it will trigger the hiding mechanism.

https://github.com/user-attachments/assets/ddf64b0e-5bdc-42d8-bf35-3ab6fce5e89a

#### Other Changes:

This work also includes some minor additions, refactors and fixes as follows:

- Added a `useLocalStorage` hook used for retaining state of split layout. It can also be used moving forward in a project if needed.
- A `FlexContentPlaceholder` was added for convenience in stories and development to fill an area for visibility. This is used in examples and stories.
- Due to issues with the background color and gradient not always reaching the bottom of the page, flex rules have been revised down to the `html`, `body` and `#root` CSS selectors with the background being moved for consistent and expected behaviours.
- As part of the above, theme colors were split to their own `variables/Colors.ts` file.
- Also, a `ThemeHelpers.ts` file was created with a `BaseStyles` styled-components `css` snippet so it can be easily imported in to the project to ensure the base rules are correctly set.
- There is a small fix to `dashboard` and `fullscreen` layout's `max-width`. It is now set to `none` to allow these to fully span the viewport as intended.
- `Layouts` contains some sample content changes.
- I've also moved a styled component out of the React component as it shouldn't have been in there. This may appear in another component to in the PR.

 ### Reviewing/Testing steps
This can be seen in Examples > Split Layouts
And also in Storybooks: Global > Organisms > Split Layout / Split Layout Nested

## Future Considerations
- There are currently three stories for layouts that are not fully featured. This was due to Storybook freezing in the browser when trying to implement Knobs controls. These should be revisited when upgrading to the newer Storybook Controls.

### CODE REVIEW

File: packages/ui-lib/src/Layouts/molecules/SplitLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+    }
+  `}
+
+`;
+
+
+const Container = styled.section<{$initialised?: 'true' | 'false', $layout?: LayoutType, $reverse?: string}>`
+  box-sizing: border-box;
+  /* width: 100vw; */
+  /* height: 100vh; */
```
Comment (isacoder) [inline packages/ui-lib/src/Layouts/molecules/SplitLayout.tsx:146]: Are these still needed?

File: packages/ui-lib/src/theme/variables/Colors.ts [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+    --white: var(--white-1);
+
+    --spinner-danger: var(--warning-8);
+    --spinner-secondary: var(--grey-8);
+    --spinner-primary: var(--primary-6);
+    --spinner-simple: var(--grey-a8);
+
+    /* --main-background-gradient: linear-gradient(180.00deg, var(--grey-2) 0%, var(--grey-3) 100%);
+    --main-background-color: var(--grey-3); */
+  }
```
Comment (isacoder) [inline packages/ui-lib/src/theme/variables/Colors.ts:543]: Is the main background variable here not longer needed?

**Review by isacoder — APPROVED** [review]
Hello Leonard,

I had previously the opportunity to review the code of the SplitLayout in another instance you shared with me. All looks right and working great. I have only added some nitpicks of code cleaning but after clearing those this will be ready to merge.

File: packages/ui-lib/src/Global/atoms/Layout.tsx [hunk 3]
```diff
@@ -7,8 +7,7 @@ export const MOBILE_NAVBAR_HEIGHT = 68;
 export const Layout = styled.div`
   display: flex;
   flex-direction: row;
-  height: 100%;
-  ${({ theme }) => theme.styles.global.background};
+  flex: 1;
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/src/Global/atoms/Layout.tsx:10]: Double checked by adding some extra content to the legacy version that's still in examples and it all looks good.

File: packages/storybook/src/stories/Global/organisms/NestedSplitLayout.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+export const _SplitLayoutNested = () => {
+
+  const reverse = false;
+
+  const nestedSplitChild = <SplitLayout 
+    layout='vertical'
+    persist 
+    persistenceKey='my_nested_key'
+    reverse={reverse} 
+    mainArea={{ content: <FlexContentPlaceholder title='Area sA' />, minSize: 120 }}
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/storybook/src/stories/Global/organisms/NestedSplitLayout.stories.tsx:37]: Fixed.

File: packages/ui-lib/src/Layouts/atoms/ResizeLine.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import styled, { css } from "styled-components";
+
+export type TResizeLineStates = 'default' | 'arrow';
+export type TResizeLineLayouts = 'horizontal' | 'vertical';
+export type TResizeLineDirection = 'up' | 'down' | 'left' | 'right' | undefined;
+
+const Inner = styled.div`
+  display: flex;
+  /* flex-direction: column; */
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/src/Layouts/atoms/ResizeLine.tsx:10]: I think it's good to remove. Fixed!

File: packages/example/src/pages/SplitLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,213 @@
+import React, { FC, useState } from "react";
+import { ThemeProvider } from 'styled-components';
+import { GlobalUI, defaultTheme, useThemeToggle, ContentLayout, SplitLayout, FlexContentPlaceholder } from "scorer-ui-kit";
+
+const SplitLayouts: FC = () => {
+
+  const {onThemeToggle, isLightMode} = useThemeToggle();
+  
+  // const layoutInstanceRef = useRef<ISplitLayoutHandles>(null);
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/example/src/pages/SplitLayout.tsx:9]: Was a little uncertain of if it was being used for something I couldn't see but I don't think it is... removed.

File: packages/example/src/pages/SplitLayout.tsx [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+            {
+              href: "/user/payments",
+              text: "Payments",
+            },
+          ]}
+        >
+          
+          <ContentLayout layout="dashboard">
+            <SplitLayout 
+              // ref={layoutInstanceRef}
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/example/src/pages/SplitLayout.tsx:199]: And removed.

File: packages/example/src/pages/Layouts.tsx [hunk 8]
```diff
@@ -5,17 +5,17 @@ import { GlobalUI, defaultTheme, PageHeader, useThemeToggle, ContentLayout, Tab,
 import { IHeaderContent } from "scorer-ui-kit/dist/Layouts";
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/example/src/pages/Layouts.tsx:5]: Fixed.

File: packages/storybook/src/stories/Global/organisms/SplitLayout.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    reverse={reverse} 
+    mainArea={{ content: <FlexContentPlaceholder title='Area A' />, minSize: 120 }} 
+    sideArea={{ content: <FlexContentPlaceholder title='Area B' />, collapsable: true, minSize: 200 }} />
+
+  const nestedSplitChild = <SplitLayout 
+    layout='vertical'
+    persist 
+    persistenceKey='my_nested_key'
+    reverse={reverse} 
+    mainArea={{ content: <FlexContentPlaceholder title='Area sA' />, minSize: 120 }}
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/storybook/src/stories/Global/organisms/SplitLayout.stories.tsx:47]: Sorted

File: packages/ui-lib/src/Layouts/molecules/SplitLayout.tsx (hunk 1 above)
Comment (atomworks) [reply packages/ui-lib/src/Layouts/molecules/SplitLayout.tsx:146]: Removed

File: packages/ui-lib/src/theme/variables/Colors.ts (hunk 2 above)
Comment (atomworks) [reply packages/ui-lib/src/theme/variables/Colors.ts:543]: I think it should be here. I've uncommented it and in the next PR, this should be moved anyway.

File: packages/storybook/src/stories/Global/organisms/SplitLayout.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    persist 
+    persistenceKey='my_unique_layout_key'
+    reverse={reverse} 
+    mainArea={{ content: nestedSplitChild, minSize: 120 }} 
+    sideArea={{ content: <FlexContentPlaceholder title='Area B' />, collapsable: true, minSize: 200 }} />
+
+  return (
+    <Container>
+      <ContentLayout layout='dashboard'>
+        {nested ? nestedSplitLayout : splitLayout};
```
Comment (isacoder) [inline packages/storybook/src/stories/Global/organisms/SplitLayout.stories.tsx:61]: Same comment about the semycolon sorry to bother you with this small thing 🙏

**Pull request merged**